.github/workflows/test.yml

```
@@ -88,5 +96,49 @@ jobs:
           cmake -S . -B build
           cmake --build build
 
+  build-python-api:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: windows-latest
+          - os: macos-latest
+          - os: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+      - uses: actions-rs/toolchain@v1
+      - name: venv作成
+        shell: bash
+        run: |
+          virtual_env="$RUNNER_TEMP"/.venv
+          python -m venv "$virtual_env"
+          if [ "$RUNNER_OS" = Windows ]; then
+            echo "$virtual_env"/Scripts >>"$GITHUB_PATH"
+          else
+            echo "$virtual_env"/bin >>"$GITHUB_PATH"
+          fi
+          echo "VIRTUAL_ENV=$virtual_env" >>"$GITHUB_ENV"
+      - shell: bash
+        run: pip install -r ./requirements.txt
+      - shell: bash
+        run: cargo build -p voicevox_core_c_api
+      - shell: bash
+        run: maturin build --locked
+      - shell: bash
+        run: maturin develop --locked
+      - name: 必要なDLLをカレントディレクトリにコピー
+        run: |
+          cp -v target/debug/build/onnxruntime-sys-*/out/onnxruntime_*/onnxruntime-*/lib/onnxruntime.dll . || true
+          cp -v target/debug/build/onnxruntime-sys-*/out/onnxruntime_*/onnxruntime-*/lib/libonnxruntime.so.* . || true
+          cp -v target/debug/build/onnxruntime-sys-*/out/onnxruntime_*/onnxruntime-*/lib/libonnxruntime.*.dylib . || true
+      - name: '`maturin develop`でインストールした`voicevox_core_python_api`を実行'
+        shell: python
+        run: import voicevox_core
```

この辺全部普通のシェルスクリプトだし、一緒にしちゃうと見通しが良いかもと思いました！
```suggestion
        shell: bash
        run: |
          # python venv
          virtual_env="$RUNNER_TEMP"/.venv
          python -m venv "$virtual_env"
          if [ "$RUNNER_OS" = Windows ]; then
            echo "$virtual_env"/Scripts >>"$GITHUB_PATH"
          else
            echo "$virtual_env"/bin >>"$GITHUB_PATH"
          fi
          echo "VIRTUAL_ENV=$virtual_env" >>"$GITHUB_ENV"
          pip install -r ./requirements.txt
          
          # maturin
          cargo build -p voicevox_core_c_api
          maturin build --locked
          run: maturin develop --locked
          
          # 必要なDLLをカレントディレクトリにコピー
          cp -v target/debug/build/onnxruntime-sys-*/out/onnxruntime_*/onnxruntime-*/lib/onnxruntime.dll . || true
          cp -v target/debug/build/onnxruntime-sys-*/out/onnxruntime_*/onnxruntime-*/lib/libonnxruntime.so.* . || true
          cp -v target/debug/build/onnxruntime-sys-*/out/onnxruntime_*/onnxruntime-*/lib/libonnxruntime.*.dylib . || true
          
          # `maturin develop`でインストールした`voicevox_core_python_api`を実行
          python -c "import voicevox_core"
```

---

あまり一つにまとめすぎてしまうとエラーが出たときの特定が難しくなると思います

---

最初に `set -eux` を付けておけば特定難度は変わらないかなと思います。
見通しが良くなる分ちょっと特定しやすくなるかもです。

---

どっちでもよさそうなんで取り込むかどうかはqryxipさんに判断を委ねようと思います

---

考えたのですが、以下の理由からそのままにしたいなと思ってます。

1. 少なくともvenv作成については、Windowsのことを考えるとjobを分離した方が扱いやすそう
2. venv作成後は(ファイルシステム以外の)変数などの状態を扱ったりする必要はないので、それだったらjobを分離した方が個人的には視認性が高そうに思える